tests/functional/cylc-combination-scripts/10-CYLC_SOURCE_DIRECTORY.t

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env bash
+# THIS FILE IS PART OF THE CYLC WORKFLOW ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#------------------------------------------------------------------------------
+# Test `cylc vip` (Validate Install Play)
+
+. "$(dirname "$0")/test_header"
+
+CYLC_ROSE=$(python -c """
```

Comment:
I'm in two minds about this?

Reasons this belongs in Cylc:
* It's testing code from Cylc.
* The Rose-stem case is just one edge case in a group and can sit with its friends.
* I only have to have one PR.

Reasons this belongs in Cylc Rose
* The test might get missed?
* Rose stem is Cylc-Rose functionality

Reply:
I would be inclined to drop the Rose part of these tests from cylc-flow as we don't have other Rose tests in cylc-flow and this is unlikely to get run.

If there are any interactions with cylc-rose that we would like to ensure are upheld (e.g. if we wanted to make `CYLC_SOURCE_DIRECTORY` available for use in the `rose-suite.conf` file [which I don't think we do?]) then we could add tests to cylc-rose for that.